In [2]:
print("Hello, DSA4153!")

Hello, DSA4153!


In [3]:
import pandas as pd
import numpy as np

print("pandas:", pd.__version__)
print("numpy:", np.__version__)

pandas: 3.0.5
numpy: 2.5.3


In [4]:
production = pd.read_csv(
    "../data/raw/Palay and Corn- Volume of Production in Metric Tons by Ecosystem-Croptype, Quarter, Semester, Region and Province, 1987-2026.csv",
    header=1
)

production.head()

,Ecosystem/Croptype,Geolocation,1987 Quarter 1,1987 Quarter 2,1987 Semester 1,1987 Quarter 3,1987 Quarter 4,1987 Semester 2,1987 Annual,1988 Quarter 1,...,2025 Quarter 4,2025 Semester 2,2025 Annual,2026 Quarter 1,2026 Quarter 2,2026 Semester 1,2026 Quarter 3,2026 Quarter 4,2026 Semester 2,2026 Annual
0,Irrigated Palay,PHILIPPINES,1514038.00,1144237.00,2658275.00,474008.00,2676825.00,3150833.00,5809108.00,1704211.00,...,4961856.71,7752046.09,15153427.75,3428108.14,3993214.05,7421322.19,..,..,..,..
1,Irrigated Palay,..CORDILLERA ADMINISTRATIVE REGION (CAR),8299.00,54856.00,63155.00,14236.00,67436.00,81672.00,144827.00,25393.00,...,111283.03,154332.97,285154.02,27985.07,101147.17,129132.24,..,..,..,..
2,Irrigated Palay,....Abra,325.00,6595.00,6920.00,131.00,9106.00,9237.00,16157.00,1379.00,...,26244.80,32307.80,51901.30,8198.58,11352.39,19550.97,..,..,..,..
3,Irrigated Palay,....Apayao,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,...,13824.79,17219.07,41739.54,2405.66,25395.66,27801.32,..,..,..,..
4,Irrigated Palay,....Benguet,105.00,1385.00,1490.00,1969.00,1762.00,3731.00,5221.00,170.00,...,4182.00,8765.25,16997.25,4834.36,2983.55,7817.91,..,..,..,..


In [5]:
production.shape

(654, 282)

In [6]:
production.columns

Index(['Ecosystem/Croptype', 'Geolocation', '1987 Quarter 1', '1987 Quarter 2',
       '1987 Semester 1', '1987 Quarter 3', '1987 Quarter 4',
       '1987 Semester 2', '1987 Annual', '1988 Quarter 1',
       ...
       '2025 Quarter 4', '2025 Semester 2', '2025 Annual', '2026 Quarter 1',
       '2026 Quarter 2', '2026 Semester 1', '2026 Quarter 3', '2026 Quarter 4',
       '2026 Semester 2', '2026 Annual'],
      dtype='str', length=282)

In [7]:
production.dtypes

Ecosystem/Croptype    str
Geolocation           str
1987 Quarter 1        str
1987 Quarter 2        str
1987 Semester 1       str
                     ... 
2026 Semester 1       str
2026 Quarter 3        str
2026 Quarter 4        str
2026 Semester 2       str
2026 Annual           str
Length: 282, dtype: object

In [8]:
production.isna().sum().sum()

np.int64(0)

In [9]:
(production == "..").sum().sum()

np.int64(17253)

In [10]:
area = pd.read_csv(
    "../data/raw/Palay and Corn- Area Harvested in Hectares by Ecosystem-Croptype, Quarter, Semester, Region and Province, 1987-2026.csv",
    header=1
)

area.head()

,Ecosystem/Croptype,Geolocation,1987 Quarter 1,1987 Quarter 2,1987 Semester 1,1987 Quarter 3,1987 Quarter 4,1987 Semester 2,1987 Annual,1988 Quarter 1,...,2025 Quarter 4,2025 Semester 2,2025 Annual,2026 Quarter 1,2026 Quarter 2,2026 Semester 1,2026 Quarter 3,2026 Quarter 4,2026 Semester 2,2026 Annual
0,Irrigated Palay,PHILIPPINES,493230.00,345450.00,838680.00,151720.00,861240.00,1012960.00,1851640.00,552640.00,...,1155524.78,1788645.90,3386903.46,773395.07,837543.69,1610938.75,..,..,..,..
1,Irrigated Palay,..CORDILLERA ADMINISTRATIVE REGION (CAR),3120.00,21050.00,24170.00,7130.00,23350.00,30480.00,54650.00,9290.00,...,26994.20,39027.20,70612.28,7142.98,24409.15,31552.13,..,..,..,..
2,Irrigated Palay,....Abra,240.00,3210.00,3450.00,140.00,5030.00,5170.00,8620.00,820.00,...,7747.00,9899.00,14591.00,1893.00,2799.00,4692.00,..,..,..,..
3,Irrigated Palay,....Apayao,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,...,3750.00,4652.00,11182.00,582.70,6210.15,6792.85,..,..,..,..
4,Irrigated Palay,....Benguet,70.00,910.00,980.00,1300.00,1040.00,2340.00,3320.00,140.00,...,1560.00,3175.00,5897.00,1621.00,1101.00,2722.00,..,..,..,..


In [11]:
area.shape

(654, 282)

In [12]:
area.columns

Index(['Ecosystem/Croptype', 'Geolocation', '1987 Quarter 1', '1987 Quarter 2',
       '1987 Semester 1', '1987 Quarter 3', '1987 Quarter 4',
       '1987 Semester 2', '1987 Annual', '1988 Quarter 1',
       ...
       '2025 Quarter 4', '2025 Semester 2', '2025 Annual', '2026 Quarter 1',
       '2026 Quarter 2', '2026 Semester 1', '2026 Quarter 3', '2026 Quarter 4',
       '2026 Semester 2', '2026 Annual'],
      dtype='str', length=282)

In [13]:
area.dtypes

Ecosystem/Croptype    str
Geolocation           str
1987 Quarter 1        str
1987 Quarter 2        str
1987 Semester 1       str
                     ... 
2026 Semester 1       str
2026 Quarter 3        str
2026 Quarter 4        str
2026 Semester 2       str
2026 Annual           str
Length: 282, dtype: object

In [14]:
area.isna().sum().sum()

np.int64(0)

In [15]:
(area == "..").sum().sum()

np.int64(17262)

In [16]:
production["Ecosystem/Croptype"].nunique()

6

In [17]:
area["Ecosystem/Croptype"].nunique()

6

In [18]:
production["Ecosystem/Croptype"].unique()

<StringArray>
['Irrigated Palay',   'Rainfed Palay',           'Palay',      'White Corn',
     'Yellow Corn',            'Corn']
Length: 6, dtype: str

In [19]:
area["Ecosystem/Croptype"].unique()

<StringArray>
['Irrigated Palay',   'Rainfed Palay',           'Palay',      'White Corn',
     'Yellow Corn',            'Corn']
Length: 6, dtype: str

In [20]:
production["Geolocation"].nunique()

109

In [21]:
area["Geolocation"].nunique()

109

## Initial Data Inspection

The group conducted an initial inspection of the two datasets using Python and pandas. The purpose of this inspection was to understand the structure, contents, and basic data quality of the datasets before proceeding with data cleaning and integration.

First, both datasets were loaded into pandas using `pd.read_csv()`. The `header=1` parameter was used because the first row of each CSV file contains the dataset title, while the second row contains the actual column names. The datasets were then inspected using `.head()` to confirm that the data was loaded correctly.

The Production dataset contains 654 rows and 282 columns, while the Area Harvested dataset also contains 654 rows and 282 columns. Both datasets contain data covering 1987 to 2026. The first two columns are `Ecosystem/Croptype` and `Geolocation`, followed by quarterly, semester, and annual values for each year.

The data types were also checked using `.dtypes`. All 282 columns in both datasets were initially read as strings. Although no actual `NaN` values were detected, the datasets contain `..` entries that represent unavailable or missing values in the source data. The Production dataset contains 17,253 `..` entries, while the Area Harvested dataset contains 17,262.

The group also examined the values of `Ecosystem/Croptype` and `Geolocation` to identify possible fields for integrating the two datasets. Both datasets contain the same six crop/ecosystem categories and 109 unique geolocation values. These matching fields provide a possible basis for connecting the Production and Area Harvested datasets during the later integration stage.

Overall, the initial inspection helped the group understand the structure of the datasets and identify data quality issues that will need to be considered during the cleaning and integration stages.

In [22]:
production_keys = production[["Ecosystem/Croptype", "Geolocation"]].drop_duplicates()

area_keys = area[["Ecosystem/Croptype", "Geolocation"]].drop_duplicates()

print("Production keys:", len(production_keys))
print("Area Harvested keys:", len(area_keys))

Production keys: 654
Area Harvested keys: 654


In [23]:
common_keys = production_keys.merge(
    area_keys,
    on=["Ecosystem/Croptype", "Geolocation"],
    how="inner"
)

print("Matching keys:", len(common_keys))

Matching keys: 654


In [24]:
unmatched_production = production_keys.merge(
    area_keys,
    on=["Ecosystem/Croptype", "Geolocation"],
    how="left",
    indicator=True
)

unmatched_production["_merge"].value_counts()

_merge
both          654
left_only       0
right_only      0
Name: count, dtype: int64

## Data Validation

The purpose of this validation was to check whether the two datasets could be reliably connected using `Ecosystem/Croptype` and `Geolocation`. These fields were examined because both datasets contain the same six crop/ecosystem categories and 109 geolocation values.

The combination of `Ecosystem/Croptype` and `Geolocation` produced 654 unique key combinations in both datasets. The validation showed that all 654 combinations were present in both the Production and Area Harvested datasets, with 0 unmatched combinations.

This indicates that the two datasets have consistent key combinations and that there are no missing matching records based on these two fields. This provides a suitable basis for the group to proceed with the planned data integration during the later stages of the project.

In [25]:
production.eq("..").sum().sort_values(ascending=False).head(20)

2026 Semester 2    654
2026 Quarter 4     654
2026 Annual        654
2026 Quarter 3     654
1987 Annual         54
1987 Quarter 1      54
1987 Quarter 2      54
1990 Quarter 1      54
1992 Semester 1     54
2022 Quarter 2      54
2022 Semester 1     54
2022 Quarter 3      54
2022 Quarter 4      54
2022 Semester 2     54
2022 Annual         54
2023 Quarter 1      54
2023 Quarter 2      54
2023 Semester 1     54
2023 Quarter 3      54
2023 Quarter 4      54
dtype: int64

In [26]:
area.eq("..").sum().sort_values(ascending=False).head(20)

2026 Semester 2    654
2026 Quarter 4     654
2026 Annual        654
2026 Quarter 3     654
1987 Annual         54
1987 Quarter 1      54
1987 Quarter 2      54
1990 Quarter 1      54
1992 Semester 1     54
2022 Quarter 2      54
2022 Semester 1     54
2022 Quarter 3      54
2022 Quarter 4      54
2022 Semester 2     54
2022 Annual         54
2023 Quarter 1      54
2023 Quarter 2      54
2023 Semester 1     54
2023 Quarter 3      54
2023 Quarter 4      54
dtype: int64

In [27]:
production[production.eq("..").any(axis=1)].head(10)

,Ecosystem/Croptype,Geolocation,1987 Quarter 1,1987 Quarter 2,1987 Semester 1,1987 Quarter 3,1987 Quarter 4,1987 Semester 2,1987 Annual,1988 Quarter 1,...,2025 Quarter 4,2025 Semester 2,2025 Annual,2026 Quarter 1,2026 Quarter 2,2026 Semester 1,2026 Quarter 3,2026 Quarter 4,2026 Semester 2,2026 Annual
0,Irrigated Palay,PHILIPPINES,1514038.00,1144237.00,2658275.00,474008.00,2676825.00,3150833.00,5809108.00,1704211.00,...,4961856.71,7752046.09,15153427.75,3428108.14,3993214.05,7421322.19,..,..,..,..
1,Irrigated Palay,..CORDILLERA ADMINISTRATIVE REGION (CAR),8299.00,54856.00,63155.00,14236.00,67436.00,81672.00,144827.00,25393.00,...,111283.03,154332.97,285154.02,27985.07,101147.17,129132.24,..,..,..,..
2,Irrigated Palay,....Abra,325.00,6595.00,6920.00,131.00,9106.00,9237.00,16157.00,1379.00,...,26244.80,32307.80,51901.30,8198.58,11352.39,19550.97,..,..,..,..
3,Irrigated Palay,....Apayao,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,...,13824.79,17219.07,41739.54,2405.66,25395.66,27801.32,..,..,..,..
4,Irrigated Palay,....Benguet,105.00,1385.00,1490.00,1969.00,1762.00,3731.00,5221.00,170.00,...,4182.00,8765.25,16997.25,4834.36,2983.55,7817.91,..,..,..,..
5,Irrigated Palay,....Ifugao,3198.00,12222.00,15420.00,2287.00,8922.00,11209.00,26629.00,9220.00,...,5344.85,12074.67,29702.82,7443.44,6629.31,14072.75,..,..,..,..
6,Irrigated Palay,....Kalinga,4125.00,29105.00,33230.00,4746.00,45124.00,49870.00,83100.00,11956.00,...,60046.10,75379.10,133721.10,4788.08,52625.50,57413.58,..,..,..,..
7,Irrigated Palay,....Mountain Province,546.00,5549.00,6095.00,5103.00,2522.00,7625.00,13720.00,2668.00,...,1640.48,8587.07,11092.00,314.95,2160.76,2475.71,..,..,..,..
8,Irrigated Palay,..REGION I (ILOCOS REGION),91932.00,46413.00,138345.00,4424.00,318416.00,322840.00,461185.00,98266.00,...,735676.50,905021.60,1402698.34,337284.77,190298.53,527583.30,..,..,..,..
9,Irrigated Palay,....Ilocos Norte,11890.00,8440.00,20330.00,1430.00,36046.00,37476.00,57806.00,10487.00,...,175012.00,222971.80,299859.80,41360.00,46887.24,88247.24,..,..,..,..


In [28]:
area[area.eq("..").any(axis=1)].head(10)

,Ecosystem/Croptype,Geolocation,1987 Quarter 1,1987 Quarter 2,1987 Semester 1,1987 Quarter 3,1987 Quarter 4,1987 Semester 2,1987 Annual,1988 Quarter 1,...,2025 Quarter 4,2025 Semester 2,2025 Annual,2026 Quarter 1,2026 Quarter 2,2026 Semester 1,2026 Quarter 3,2026 Quarter 4,2026 Semester 2,2026 Annual
0,Irrigated Palay,PHILIPPINES,493230.00,345450.00,838680.00,151720.00,861240.00,1012960.00,1851640.00,552640.00,...,1155524.78,1788645.90,3386903.46,773395.07,837543.69,1610938.75,..,..,..,..
1,Irrigated Palay,..CORDILLERA ADMINISTRATIVE REGION (CAR),3120.00,21050.00,24170.00,7130.00,23350.00,30480.00,54650.00,9290.00,...,26994.20,39027.20,70612.28,7142.98,24409.15,31552.13,..,..,..,..
2,Irrigated Palay,....Abra,240.00,3210.00,3450.00,140.00,5030.00,5170.00,8620.00,820.00,...,7747.00,9899.00,14591.00,1893.00,2799.00,4692.00,..,..,..,..
3,Irrigated Palay,....Apayao,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,...,3750.00,4652.00,11182.00,582.70,6210.15,6792.85,..,..,..,..
4,Irrigated Palay,....Benguet,70.00,910.00,980.00,1300.00,1040.00,2340.00,3320.00,140.00,...,1560.00,3175.00,5897.00,1621.00,1101.00,2722.00,..,..,..,..
5,Irrigated Palay,....Ifugao,910.00,5110.00,6020.00,960.00,2760.00,3720.00,9740.00,2490.00,...,1677.20,3769.20,8748.50,1943.00,2749.00,4692.00,..,..,..,..
6,Irrigated Palay,....Kalinga,1590.00,9040.00,10630.00,1850.00,13040.00,14890.00,25520.00,4480.00,...,11736.00,14859.00,26726.00,992.00,10875.00,11867.00,..,..,..,..
7,Irrigated Palay,....Mountain Province,310.00,2780.00,3090.00,2880.00,1480.00,4360.00,7450.00,1360.00,...,524.00,2673.00,3467.78,111.28,675.00,786.28,..,..,..,..
8,Irrigated Palay,..REGION I (ILOCOS REGION),35660.00,15650.00,51310.00,2220.00,115630.00,117850.00,169160.00,35150.00,...,151820.00,185311.86,286605.14,66476.83,38471.00,104947.83,..,..,..,..
9,Irrigated Palay,....Ilocos Norte,4890.00,3050.00,7940.00,680.00,17530.00,18210.00,26150.00,3910.00,...,31638.00,40636.00,54812.00,7520.00,8965.00,16485.00,..,..,..,..
